src/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -227,6 +239,37 @@ internal static int GetRootLength(string path)
             return i;
         }
 
+        /// <summary>
+        /// Returns true if the path specified is relative to the current drive or working directory.
+        /// Returns false if the path is fixed to a specific drive or UNC path.  This method does no
+        /// validation of the path (URIs will be returned as relative as a result).
+        /// </summary>
+        /// <remarks>
+        /// Handles paths that use the alternate directory separator.  It is a frequent mistake to
+        /// assume that rooted paths (Path.IsPathRooted) are not relative.  This isn't the case.
+        /// </remarks>
+        internal static bool IsPathRelative(string path)
+        {
+            if (path.Length < 2)
+            {
+                // It isn't fixed, it must be relative.  There is no way to specify a fixed
+                // path with one character (or less).
+                return true;
+            }
+
+            if ((path[0] == '\\') || (path[0] == '/'))
```

Comment:
We should only do this for perf if we know that there is a problem, otherwise we should go for readability and consistency. At a minimum we can use the constants so that intentions are clearer.